### ISSUE TITLE
Blade template syntax highlighting

### ISSUE THREAD

**Issue by web-engineer** [nbphpcouncil/nb-laravel-plugin#9]
Question for you - are there plans to support blade syntax highlighting with this plugin, this was the main reason why I looked to a plugin for laravel support assuming this would be a significant area of improvement. 

I notice [laravel4-blade](https://github.com/RobinRadic/netbeans-laravel4-blade/issues) was started to cover this but it doesn't look like the author got very far with their efforts.

**Comment by junichi11** [comment]
> Question for you - are there plans to support blade syntax highlighting with this plugin, this was the main reason why I looked to a plugin for laravel support assuming this would be a significant area of improvement.

@jtakakura Do you support this?

@web-engineer I've looked at the source code of the plugin. Probably, That plugin just provides support for `comment`. Others, do nothing at the moment.

**Comment by web-engineer** [comment]
I'd be happy to help with this if I can, would really complete the package since netbeans is pretty dreadful for editing blade files presently and getting insight at that level would be awesome!

**Comment by swissmant** [comment]
It would be great if you are able to add this. It is pretty frustrating editing blade templates using NetBeans right now. Thanks!

**Comment by Monomachus** [comment]
@tmysik This is the issue I was talking about in the email

**Comment by web-engineer** [comment]
Thank you for your email, I'm currently on leave and will be back on Tuesday 28th April. 

If your in urgent need of assistance please telephone the office on 01392 253304.

Many thanks

**Comment by tmysik** [comment]
I see one major problem here - Blade files are stored as PHP files, at least that is my undestanding based on [1]. This is "wrong" since the Blade content apparently is NOT a PHP file (much better would be to use a new file extension, `.blade` for example). In other words, I am not sure whether will be possible to create embedding for Blade into PHP MIME type (PHP files) in NetBeans.

Thanks.
[1] http://laravel.com/docs/4.2/templates

**Comment by Monomachus** [comment]
@tmysik Can't we somehow tell NetBeans that everything with .blade.php should be processed as an extension that is different from .php?

**Comment by tmysik** [comment]
I am not sure but it might work, maybe.

**Comment by junichi11** [comment]
Perhaps, if we use MIMEResolver[1], it will be resolved.(Sorry if I'm wrong) But I think that we have to learn many things if we implement this feature. e.g. how to use lexer and parser (jflex, antlr, e.t.c.). Need many time for it.

[1] http://bits.netbeans.org/dev/javadoc/org-openide-filesystems/org/openide/filesystems/MIMEResolver.html

**Comment by tmysik** [comment]
Yes, using MIME resolver with "high" (in fact, low) position will do that. But still, there will be troubles with the PHP + embedding IMO.

**Comment by junichi11** [comment]
@tmysik Thanks for your comment :smile:

**Comment by Chris2011** [comment]
NetBeans or this plugin needs full laravel support. That mean *.blade.php file extension should treated as a new mimetype like text/laravel+php+html or smth like that. Like we have it for twig. And Syntax highlighting for the blade syntax. Any other features can be found here: https://blog.jetbrains.com/phpstorm/2014/08/blade-template-engine-support-in-phpstorm-8-eap/

This is really missing.

**Comment by haidubogdan** [comment]
Hi all,
I started working on a blade plugin : https://github.com/haidubogdan/netbeans-php-blade-plugin
It is still experimental, and it works just for Netbeans 12.3 and 12.5. 
I will have to see how I can adapt for previous versions.

**Comment by Chris2011** [comment]
@haidubogdan great news. Just let us know, what we can do here to help you :). I will just test it as soon as I can.
